ksql-engine/src/test/java/io/confluent/ksql/codegen/SqlToJavaVisitorTest.java

```diff
@@ -127,4 +127,23 @@ public void shouldGenerateCorrectCodeForComparisonWithNegativeNumbers() {
     assertThat(javaExpression, equalTo("((((Object)(TEST1_COL3)) == null || ((Object)(-10.0)) == null) ? false : (TEST1_COL3 > -10.0))"));
   }
 
+  @Test
```

Comment:
I would unit test the three cases separately:
starting wildcard, ending wildcard, start and ending wildcard.